activemq-unit-tests/src/test/java/org/apache/activemq/jms2/ActiveMQJMS2MessageTypesTest.java

```diff
@@ -0,0 +1,582 @@
+
+            List<String> sentMessageIds = new LinkedList<>();
+            sentMessageIds.add(ActiveMQJMS2TestSupport.sendMessage(jmsContext, destination, message, null, null, null, null, jmsCorrelationID, jmsReplyTo, jmsType, null, null));
+
+            Message recvMessage = null;
+            List<Message> recvMessages = new LinkedList<>();
+            int loopCount = 0;
+            int maxLoops = 10;
+            boolean done = false;
+            do {
+                recvMessage = jmsConsumer.receive(500l);
+                if(recvMessage != null) {
+                    recvMessages.add(recvMessage);
+                }
+
+                if(recvMessages.size() == 1) {
+                    done = true;
+                }
+                loopCount++;
+            } while (loopCount <= maxLoops && !done);
+
+            int validatedCount = 0;
+            for(javax.jms.Message tmpMessage : recvMessages) {
+                ActiveMQJMS2TestSupport.validateMessageNonQOSHeaders(tmpMessage, messageType, messagePayload, jmsCorrelationID, jmsReplyTo, jmsType);
+                validatedCount++;
+            }
+            assertEquals(Integer.valueOf(1), Integer.valueOf(validatedCount));
+            jmsConsumer.close();
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }   
+
+    @Test
+    public void testMessageDisableMessageID() {
+
+        try(JMSContext jmsContext = activemqConnectionFactory.createContext("admin", "admin", Session.AUTO_ACKNOWLEDGE)) {
+            assertNotNull(jmsContext);
+            Destination destination = ActiveMQJMS2TestSupport.generateDestination(jmsContext, destinationType, methodNameDestinationName);
+            assertNotNull(destination);
+            JMSConsumer jmsConsumer = jmsContext.createConsumer(destination);
+            assertNotNull(jmsConsumer);
+            jmsContext.start();
+
+            Message message = ActiveMQJMS2TestSupport.generateMessage(jmsContext, messageType, messagePayload);
+            String jmsMessageID = (ActiveMQJMS2TestSupport.sendMessage(jmsContext, destination, message, null, null, true, null, null, null, null, null, null));
+
+            Message recvMessage = null;
+            List<Message> recvMessages = new LinkedList<>();
+            int loopCount = 0;
+            int maxLoops = 10;
+            boolean done = false;
+            do {
+                recvMessage = jmsConsumer.receive(500l);
+                if(recvMessage != null) {
+                    recvMessages.add(recvMessage);
+                }
+
+                if(recvMessages.size() == 1) {
+                    done = true;
+                }
+                loopCount++;
+            } while (loopCount <= maxLoops && !done);
+
+            int validatedCount = 0;
+            for(javax.jms.Message tmpMessage : recvMessages) {
+                ActiveMQJMS2TestSupport.validateMessageDisableMessageID(tmpMessage, messageType, messagePayload, jmsMessageID);
```

Comment:
What is this validating exactly? That it actually has a messageID since disableMessageID isnt supported?

Reply:
Yes, for completeness against the spec changes. This is negative testing the optional DisableMessageID support. Also, in the (unlikely) event that an effort comes along to add support for messageID-less delivery these will fail.